### PR TITLE
Kostik/fix account edit workflow/aact 559

### PR DESCRIPTION
**Pull request by kostik700015**
That PR fixes account edit workflow.

Changes applied: 

1. “Database ID” changed to “Database Username”

2. Added a field for email.

3. When we have a success message indicating the user’s information was updated successfully.

4. The password & pass_confirmation fields removed from the devise/registrations/_form.html.erb template.

5. Created a new action in the registrations controller password with a view that is identical to _form.html.erb except that it has only “New Password” “Password Confirmation” and “Current Password”, this form will only be used when the user wants to change their password. Make the route for this action /users/password

6. From /users/edit which is the Edit Profile page,  the link to this password action /users/password added.

image_2023-10-04_18-11-53
image_2023-10-04_18-11-29
image_2023-10-04_16-39-04

### CODE REVIEW

File: app/controllers/users/registrations_controller.rb [hunk 1]
```diff
@@ -29,10 +29,20 @@ def destroy
...
+  def password
+    redirect_to root_path if current_user.nil? 
+  end
+
   protected
 
-  def update_resource(resource, params)
+  def update_resource(resource, params) 
+    if params[:password_confirmation].nil? && resource.errors.size == 0
+      flash.now[:notice] = 'User succesfully updated!'
```
Comment (Javier-Jimenez-18) [inline app/controllers/users/registrations_controller.rb:40]: Keep the "green" flash notice and delete the "red" flash notice (usually means error).

File: app/controllers/users/registrations_controller.rb [hunk 2]
```diff
@@ -29,10 +29,20 @@ def destroy
...
+    end
     resource.update(params)
+    if resource.errors.size == 0 && !params[:password_confirmation].nil?
+      flash.now[:notice] = 'Password succesfully updated!'
```
Comment (Javier-Jimenez-18) [inline app/controllers/users/registrations_controller.rb:44]: Same as above, keep the "green" success message and delete the "red" success message.

**Review by Javier-Jimenez-18 — CHANGES_REQUESTED** [review]
Good job. Just a couple of small changes. When you make the changes, provide a screenshot of the view page for us to see how it looks.

**Review by kholtzman5610 — CHANGES_REQUESTED** [review]
I agree with Javier suggestions definitely do those if you haven't already. 

Only thing I would suggest to do is to get the check to pass too. The test that relates to this PR would be this one below:

Failures:
1) Users Sign Up Page sign up - verify new user validations
     Failure/Error: expect(page).to have_field 'user_password'
       expected to find field "user_password" that is not disabled but there were no matches
      ./spec/features/users_sign_up_page_spec.rb:160:in `block (2 levels) in <top (required)>'

Failed examples:
rspec ./spec/features/users_sign_up_page_spec.rb:5 # Users Sign Up Page sign up - verify new user validations

**Comment by kostik700015** [comment]
> I agree with Javier suggestions definitely do those if you haven't already.
> 
> Only thing I would suggest to do is to get the check to pass too. The test that relates to this PR would be this one below:
> 
> Failures:
> 
> 1. Users Sign Up Page sign up - verify new user validations
>    Failure/Error: expect(page).to have_field 'user_password'
>    expected to find field "user_password" that is not disabled but there were no matches
>    ./spec/features/users_sign_up_page_spec.rb:160:in `block (2 levels) in <top (required)>'
> 
> Failed examples: rspec ./spec/features/users_sign_up_page_spec.rb:5 # Users Sign Up Page sign up - verify new user validations

Thank you for reviewing, everything is fixed.
image_2023-10-09_12-38-30

**Comment by kostik700015** [comment]
> Good job. Just a couple of small changes. When you make the changes, provide a screenshot of the view page for us to see how it looks.

Thank you for reviewing, everything is fixed.
image_2023-10-09_12-38-30

**Pull request merged**